Makefile

```diff
@@ -0,0 +1,165 @@
+# NOTE: This will take 1+ hours to run and the PUDL DB will not be retained.
+local-pytest-ci-all-years: docs-clean
+	${coverage_erase}
+	doc8 docs/ README.rst
+	coverage run ${covargs} -- ${CONDA_PREFIX}/bin/sphinx-build -W -b html docs docs/_build/html
+	pytest ${pytest_args} --doctest-modules src/pudl test/unit
+	pytest ${pytest_args} --etl-settings ${etl_full_yml} test/integration
+	${coverage_report}
+
+# Run the full ETL, generating new FERC & PUDL SQLite DBs and EPA CEMS Parquet files.
+# Then run the full integration tests and data validations on all years of data.
+# NOTE: This will clobber your existing databases and takes hours to run!!!
+# Backgrounding the data validation and integration tests and using wait allows them to
+# run in parallel.
+nuke: docs-clean
+	${coverage_erase}
+	doc8 docs/ README.rst
+	coverage run ${covargs} -- ${CONDA_PREFIX}/bin/sphinx-build -W -b html docs docs/_build/html
+	pytest ${pytest_args} --doctest-modules src/pudl test/unit
+	pytest ${pytest_args} \
+		--etl-settings ${etl_fast_yml} \
+		test/integration
+	rm -f tox-nuke.log
+	coverage run ${covargs} -- \
+		src/pudl/convert/ferc_to_sqlite.py \
+		--logfile tox-nuke.log \
+		--clobber \
+		${gcs_cache_path} \
+		${etl_full_yml}
+	coverage run ${covargs} -- \
+		src/pudl/cli/etl.py \
+		--logfile tox-nuke.log \
+		--clobber \
+		${gcs_cache_path} \
+		${etl_full_yml}
+	pudl_check_fks
+	pytest ${pytest_args} --live-dbs --etl-settings ${etl_full_yml} test/integration & \
+	pytest ${pytest_args} --live-dbs test/validate & \
+	wait
+	${coverage_report}
+
+########################################################################################
+# Some miscellaneous test cases
+########################################################################################
+
+# Check that designated Jupyter notebooks can be run against the current DB
+pytest-jupyter:
+	pytest --live-dbs test/integration/jupyter_notebooks_test.py
+
+# Compare actual and expected number of rows in many tables:
+pytest-minmax-rows:
+	pytest --live-dbs \
+		test/validate/epacamd_eia_test.py::test_minmax_rows \
+		test/validate/ferc1_test.py::test_minmax_rows \
+		test/validate/eia_test.py::test_minmax_rows \
+		test/validate/mcoe_test.py::test_minmax_rows_mcoe
+
+# Build the FERC 1 and PUDL DBs, ignoring foreign key constraints.
+# Identify any plant or utility IDs in the DBs that haven't yet been mapped
+# NOTE: This probably needs to be turned into a script of some kind not a test.
+# In particular, building these DBs without checking FK constraints, in a location
+# where they aren't going to clobber existing user DBs
+unmapped-ids:
+	pytest \
+		--save-unmapped-ids \
+		--ignore-foreign-key-constraints \
+		--etl-settings ${etl_full_yml} \
+		test/integration/glue_test.py
+
+########################################################################################
+# The github- prefixed targets are meant to be run by GitHub Actions
```

Comment:
Could you elaborate on why there are different commands for local and github in this comment?

Reply:
For the GitHub ones I'm having them install PUDL and deal with coverage before & after the run. But maybe we could do those things every time locally or on GitHub and it would be fine?

Reply:
Got it. I say we keep it as is for now. Just run coverage on GitHub.

Reply:
I think it would be better if the `make` targets are exactly what we do on CI - once we start having `local-`, `github-`, and targets with no prefix we lose the standardization benefit.

This would also let us have less code in the Makefile which is a plus.

The tricky thing is handling PUDL installation & coverage.

PUDL installation: this should be part of a separate `devenv` target, I think - something like

```make
devenv: conda-env
        mamba run -n pudl-dev pip install -e .

conda-env: environments/conda-lock.yml
        conda-lock install -n pudl-dev $^
```

Coverage: on CI, we're doing coverage for unit/integration/docs separately, and combining them at the end. Locally, we might want to do coverage for any subset of those, which is the impetus behind the various `local-pytest-ci` targets. I think we should:
* stop running `coverage erase` within `make` - on CI runners it is unnecessary, and locally this forces a new target for every subset of tests we want to run coverage for.
* include the coverage args for all unit/integration/doc-build targets, for local/CI consistency. If you really want to skip coverage, you can always run the naked command.

Here's what I have locally:

```diff
diff --git a/Makefile b/Makefile
index a86b30874..246ee37f5 100644
--- a/Makefile
+++ b/Makefile
@@ -14,6 +14,12 @@ pip_install_pudl := pip install --no-deps --editable ./
 dagster:
 	dagster dev -m pudl.etl -m pudl.ferc_to_sqlite
 
+devenv: conda-env
+	mamba run -n pudl-dev pip install -e .
+
+conda-env: environments/conda-lock.yml
+	conda-lock install -n pudl-dev $^
+
 ########################################################################################
 # Conda lockfile generation
 ########################################################################################
@@ -47,16 +53,19 @@ docs-clean:
 
 docs-build: docs-clean
 	doc8 docs/ README.rst
-	sphinx-build -W -b html docs docs/_build/html
+	coverage run ${covargs} -- ${CONDA_PREFIX}/bin/sphinx-build -W -b html docs docs/_build/html
 
 ########################################################################################
 # Generic pytest commands for local use, without test coverage
 ########################################################################################
 pytest-unit:
-	pytest --doctest-modules src/pudl test/unit
+	pytest ${pytest_args} --doctest-modules src/pudl test/unit
 
 pytest-integration:
-	pytest test/integration
+	pytest ${pytest_args} --etl-settings ${etl_fast_yml} test/integration
+
+pytest-integration-full:
+	pytest ${pytest_args} --etl-settings ${etl_full_yml} test/integration
 
 pytest-validate:
 	pytest --live-dbs test/validate
@@ -65,41 +74,17 @@ pytest-validate:
 ########################################################################################
 # More complex pytest commands for local use that collect test coverage
 ########################################################################################
-# Run unit & integration tests on 1-2 years of data and collect test coverage data.
-local-pytest-ci: docs-clean
-	${coverage_erase}
-	doc8 docs/ README.rst
-	coverage run ${covargs} -- ${CONDA_PREFIX}/bin/sphinx-build -W -b html docs docs/_build/html
-	pytest ${pytest_args} --doctest-modules src/pudl test/unit
-	pytest ${pytest_args} --etl-settings ${etl_fast_yml} test/integration
-	${coverage_report}
-
-# Run unit & integration tests on ALL years of data and collect test coverage data.
-# NOTE: This will take 1+ hours to run and the PUDL DB will not be retained.
-local-pytest-ci-all-years: docs-clean
-	${coverage_erase}
-	doc8 docs/ README.rst
-	coverage run ${covargs} -- ${CONDA_PREFIX}/bin/sphinx-build -W -b html docs docs/_build/html
-	pytest ${pytest_args} --doctest-modules src/pudl test/unit
-	pytest ${pytest_args} --etl-settings ${etl_full_yml} test/integration
-	${coverage_report}
-
 # Run the full ETL, generating new FERC & PUDL SQLite DBs and EPA CEMS Parquet files.
 # Then run the full integration tests and data validations on all years of data.
 # NOTE: This will clobber your existing databases and takes hours to run!!!
 # Backgrounding the data validation and integration tests and using wait allows them to
 # run in parallel.
-nuke: docs-clean
-	${coverage_erase}
-	doc8 docs/ README.rst
-	coverage run ${covargs} -- ${CONDA_PREFIX}/bin/sphinx-build -W -b html docs docs/_build/html
-	pytest ${pytest_args} --doctest-modules src/pudl test/unit
-	pytest ${pytest_args} \
-		--etl-settings ${etl_fast_yml} \
-		test/integration
+# 
+
+nuke: docs-build pytest-unit
 	rm -f tox-nuke.log
 	coverage run ${covargs} -- \
-		src/pudl/convert/ferc_to_sqlite.py \
+		src/pudl/ferc_to_sqlite/cli.py \
 		--logfile tox-nuke.log \
 		--clobber \
 		${gcs_cache_path} \
@@ -126,11 +111,7 @@ pytest-jupyter:
 
 # Compare actual and expected number of rows in many tables:
 pytest-minmax-rows:
-	pytest --live-dbs \
-		test/validate/epacamd_eia_test.py::test_minmax_rows \
-		test/validate/ferc1_test.py::test_minmax_rows \
-		test/validate/eia_test.py::test_minmax_rows \
-		test/validate/mcoe_test.py::test_minmax_rows_mcoe
+	pytest --live-dbs test/validate -k minmax_rows
 
 # Build the FERC 1 and PUDL DBs, ignoring foreign key constraints.
 # Identify any plant or utility IDs in the DBs that haven't yet been mapped
```

Couple extra notes:

`nuke`:
- I think `clobber` has some unpleasant behavior with `ferc_to_sqlite` where it will delete the XBRL databases when reading the 2022 files, but...
  - `src/pudl/conver/ferc_to_sqlite.py` doesn't even exist, so I think `nuke` was totally broken already
- should we be managing logfiles like this in dagster-land? seems a little clunky, but I also know logs in dagster are funky

`pytest-minmax-rows`: this is a shortcut to adding `@pytest.mark.minmax` to the `minmax_rows` tests. I sort of think this particular make target should be replaced by the built-in pytest test filtering tooling. Maybe the jupyter one too.